pinot-server/src/main/java/org/apache/pinot/server/starter/helix/BaseServerStarter.java

```diff
@@ -707,6 +710,13 @@ public void start()
     _serverQueriesDisabledTracker =
         new ServerQueriesDisabledTracker(_helixClusterName, _instanceId, _helixManager, serverMetrics);
     _serverQueriesDisabledTracker.start();
+
+    long startupDurationMs = System.currentTimeMillis() - startTimeMs;
```

Comment:
The code can be simplified. `isStartupStatusCheckGood` is not required. It should be sufficient to check `ServiceStatus.getServiceStatus(_instanceId)` and then set the right service metric.

Reply:
it can, but it's a small race condition. The server may have chosen to start without startup status being `GOOD`, but it might be `GOOD` by the time we get here. In practice, it probably doesn't matter. I don't have a strong opinion, do you prefer I keep the code simpler with your approach?

Reply:
I didnt think about the race condition and I agree that it doesnt matter. The diff will be much simpler - only the timers and if-else condition is required.

Also you will have to increment the failure metric in the catch block.